impacted-test-engine/src/main/java/com/teamscale/test_impacted/test_descriptor/JUnitPlatformSuiteDescriptorResolver.java

```diff
@@ -0,0 +1,101 @@
+
+		ITestDescriptorResolver secondaryTestDescriptorResolver = TestDescriptorResolverRegistry.getTestDescriptorResolver(
+				secondaryEngineSegments.get(0).getValue());
+		if (secondaryTestDescriptorResolver == null) {
+			LOGGER.severe(() -> "Cannot find a secondary engine nested under the junit-platform-suite engine " +
+					"(assuming structure [engine:junit-platform-suite]/[suite:mySuite]/[engine:anotherEngine])" +
+					"Using " + testDescriptor.getUniqueId().toString() + "as uniform path as fallback.");
+			return Optional.of(testDescriptor.getUniqueId().toString());
+		}
+
+		Optional<String> secondaryUniformPath = secondaryTestDescriptorResolver.getUniformPath(testDescriptor);
+		if (!secondaryUniformPath.isPresent()) {
+			LOGGER.severe(() -> "Secondary test descriptor resolver for engine " +
+					secondaryEngineSegments.get(0).getValue() + " was not able to resolve the uniform path. " +
+					"Using " + testDescriptor.getUniqueId().toString() + "as fallback.");
+			return Optional.of(testDescriptor.getUniqueId().toString());
+		}
+
+		return Optional.of(suite + "/" + secondaryUniformPath.get());
+	}
+
+	@Override
+	public Optional<String> getClusterId(TestDescriptor testDescriptor) {
+		List<UniqueId.Segment> segments = testDescriptor.getUniqueId().getSegments();
+		if (verifySegments(segments)) {
+			LOGGER.severe(
+					() -> "Assuming structure [engine:junit-platform-suite]/[suite:mySuite]/[engine:anotherEngine] " +
+							"for junit-platform-suite tests. Using "
+							+ testDescriptor.getUniqueId()
+							.toString() + "as cluster id as fallback.");
+			return Optional.of(testDescriptor.getUniqueId().toString());
+		}
+
+		String suite = segments.get(1).getValue().replace('.', '/');
+		List<UniqueId.Segment> secondaryEngineSegments = segments.subList(2, segments.size());
+
+		ITestDescriptorResolver secondaryTestDescriptorResolver = TestDescriptorResolverRegistry.getTestDescriptorResolver(
+				secondaryEngineSegments.get(0).getValue());
+		if (secondaryTestDescriptorResolver == null) {
+			LOGGER.severe(() -> "Cannot find a secondary engine nested under the junit-platform-suite engine " +
+					"(assuming structure [engine:junit-platform-suite]/[suite:mySuite]/[engine:anotherEngine])" +
+					"Using " + testDescriptor.getUniqueId().toString() + "as cluster id as fallback.");
+			return Optional.of(testDescriptor.getUniqueId().toString());
+		}
+
+		Optional<String> secondaryClusterId = secondaryTestDescriptorResolver.getClusterId(testDescriptor);
+		if (!secondaryClusterId.isPresent()) {
+			LOGGER.severe(() -> "Secondary test descriptor resolver for engine " +
+					secondaryEngineSegments.get(0).getValue() + " was not able to resolve the cluster id. " +
+					"Using " + testDescriptor.getUniqueId().toString() + "as fallback.");
+			return Optional.of(testDescriptor.getUniqueId().toString());
+		}
+		return Optional.of(suite + "/" + secondaryClusterId.get());
```

Comment:
You could extract a method that determines the delegate `ITestDescriptorResolver` and accepts a lambda `(ITestDescriptorResolver) -> Optional<String>`, which will determine the suffix that should be appended to the suite. You should be able to reuse this for both methods and can hence get rid of the clone.